### PR TITLE
Add a View > Targets menu item for non-proteomic UI modes.

### PR DESCRIPTION
**Pull request by bspratt**
Visible in Mixed and Small molecule modes only. Enabled only when the Targets window is not open.

Also, hide the View > Modifications menu item for small molecule UI mode.

This fixes:

Issue 791 Inconsistencies in Small Molecule UI mode make it difficult to reopen a closed Targets window
"From Brendan:
Bug found teaching Skyline Online. In small molecule mode, View > Proteins > etc. gets removed but not replaced with anything, which makes it confusing how to get the Targets window back if you close it. We should replace it with a View > Targets menu item that brings up the Targets view.

Also, we seem to leave View > Modifications > etc. in small molecule mode, when it doesn't apply to small molecules. It also brings up the Targets window. So, that was my easiest solution to getting the Targets window back after closing it, but the menu item should not be there. So we should be removing View > Proteins and Modifications and replacing them with View > Targets."

### CODE REVIEW

**Comment by bspratt** [comment]
I figure the only likely design controversy here is the enable/disable of the menu item based on the Targets window visibility. I think it's good that users are made aware of its availability, but it would be confusing if it was left enabled when clicking on it would produce no action.

Some screen shots (note the UI modes)

image
image
image
image
image

**Comment by brendanx67** [comment]
Don't disable it when it is visible, just put a check next to it, and when you choose it and it is visible, it should still set the focus to the Targets view. It is an "Activate" command, and not a Show/Hide command, but we do put a check by it when it is visible to let the user know that Skyline thinks it is visible. So, if it is not immediately apparent, the user might want to search for it in a situation where there is a lot of screen real estate and the window may be floating and partially off screen.

**Review by nickshulman — APPROVED** [review]
Looks good to me

**Comment by bspratt** [comment]
So:
image
image
image
image
image

File: pwiz_tools/Skyline/Menus/ViewMenu.cs [hunk 1]
```diff
@@ -827,6 +832,7 @@ public void DocumentUiChanged()
         }
         public void ViewMenuDropDownOpening()
         {
+            viewTargetsMenuItem.Checked = Settings.Default.ShowPeptides;
```
Comment (brendanx67) [inline pwiz_tools/Skyline/Menus/ViewMenu.cs:835]: It looks like the targets window can be shown without setting Settings.Default.ShowPeptides to true, which should get fixed before relying on it in this way. Look for uses of SkylineWindow.ShowSequenceTreeForm().

**Review by brendanx67 — CHANGES_REQUESTED** [review]
I think the line to set Settings.Default.ShowPeptides needs to move in the function SkylineWindow.ShowSequenceTreeForm.

File: pwiz_tools/Skyline/Menus/ViewMenu.cs (hunk 1 above)
Comment (bspratt) [reply pwiz_tools/Skyline/Menus/ViewMenu.cs:835]: Huh, ShowSequenceTreeForm(false) appears to very carefully not change the value of Settings.Default.ShowPeptides, so I guess I don't really understand what that value means, but it seems it's not a reliable indicator of that window's visibility.

I'll set that checkmark value to reflect the targets window being non-null and visible.

Comment (brendanx67) [reply pwiz_tools/Skyline/Menus/ViewMenu.cs:835]: Sounds safest. I am pretty sure we are not either hooking when the user clicks the X button to close the Targets window. So, yeah, not reliable to use the Settings value as an indicator of Targets view visibility.

**Comment by bspratt** [comment]
OK, I'll back out anything that directly messes with that settings value.

**Comment by brendanx67** [comment]
That sounds like more than I was suggesting. I just want you to put back 2 lines in your last commit. Using the View menu to show the Targets view changes the value today, and it should change the value for your View > Targets menu item. However, calls to SkylineWindow.ShowSequenceTreeForm(true) should not change the Settings.Default.ShowPeptides value, as it has not in the past. Your change to rely on visibility for the menu check mark was the right fix.

**Comment by bspratt** [comment]
Agreed, I just meant to say no *new* manipulations. I think you'll find the current code meets your expectations, including having View>Targets available in all UI modes per our side conversation.

**Pull request merged**